pkg/m3admin/namespace.go

```diff
@@ -0,0 +1,115 @@
+)
+
+const (
+	// TODO(PS) add to spec for crd?
+	// defaults for namespace creation
+	_namespaceURI                                                    = "/api/v1/namespace"
+	_defaultName                                                     = "default"
+	_defaultBootstrapEnabled                                         = true
+	_defaultFlushEnabled                                             = true
+	_defaultWritesToCommitLog                                        = true
+	_defaultCleanupEnabled                                           = true
+	_defaultSnapshotEnabled                                          = false
+	_defaultRepairEnabled                                            = false
+	_defaultRententionOptionRetentionPeriodNanos                     = int64(172800000000000)
+	_defaultRententionOptionBlockSizeNanos                           = int64(7200000000000)
+	_defaultRententionOptionBufferFutureNanos                        = int64(600000000000)
+	_defaultRententionOptionBufferPastNanos                          = int64(600000000000)
+	_defaultRententionOptionBlockDataExpiry                          = true
+	_defaultRententionOptionBlockDataExpiryAfterNotAccessPeriodNanos = int64(300000000000)
+	_defaultIndexOptionenabled                                       = true
+	_defaultIndexOptionBlockSizeNanos                                = int64(7200000000000)
+)
+
+//NamespaceCreate will create a namespace
+func (m *m3admin) NamespaceCreate(cluster *myspec.M3DBCluster) error {
+	namespaceCreateRequest := admin.NamespaceAddRequest{
+		Name: cluster.GetObjectMeta().GetName(),
+		Options: &namespace.NamespaceOptions{
+			BootstrapEnabled:  _defaultBootstrapEnabled,
+			FlushEnabled:      _defaultFlushEnabled,
+			WritesToCommitLog: _defaultWritesToCommitLog,
+			CleanupEnabled:    _defaultCleanupEnabled,
+			RepairEnabled:     _defaultRepairEnabled,
+			RetentionOptions: &namespace.RetentionOptions{
+				RetentionPeriodNanos:                     _defaultRententionOptionRetentionPeriodNanos,
+				BlockSizeNanos:                           _defaultRententionOptionBlockSizeNanos,
+				BufferFutureNanos:                        _defaultRententionOptionBufferFutureNanos,
+				BufferPastNanos:                          _defaultRententionOptionBufferPastNanos,
+				BlockDataExpiry:                          _defaultRententionOptionBlockDataExpiry,
+				BlockDataExpiryAfterNotAccessPeriodNanos: _defaultRententionOptionBlockDataExpiryAfterNotAccessPeriodNanos,
+			},
+			SnapshotEnabled: _defaultSnapshotEnabled,
+			IndexOptions: &namespace.IndexOptions{
+				Enabled:        _defaultIndexOptionenabled,
+				BlockSizeNanos: _defaultIndexOptionBlockSizeNanos,
+			},
+		},
+	}
+	data, err := json.Marshal(namespaceCreateRequest)
+	if err != nil {
+		return err
+	}
+	// TODO(PS) - Remove hardcoded references
+	url := fmt.Sprintf("http://%s.%s:7201%s", _m3coordinatorServiceName, cluster.GetNamespace(), _namespaceURI)
```

Comment:
nit: this snippet is repeated a bunch of times. Make it a helper function --

```
func (n *namespaceClient) url() string {
  ...
}
```

are you going to remove the hardcoded refs in this PR?